File: 25-notebooks/npm-package/notebooks.ts
import axios from 'axios';
const API = 'https://api.costkatana.com/api'\;
const KEY = process.env.COST_KATANA_API_KEY;

export async function createNotebook(name: string, description?: string) {
  const res = await axios.post(`${API}/notebooks`, {name, description},
    {headers: {'Authorization': `Bearer ${KEY}`}});
  console.log('✅ Notebook created:', res.data.data.id);
  return res.data.data;
}

export async function listNotebooks() {
  const res = await axios.get(`${API}/notebooks`,
    {headers: {'Authorization': `Bearer ${KEY}`}});
  console.log(`📓 Found ${res.data.data.length} notebooks`);
  return res.data.data;
}

File: 25-notebooks/python-sdk/notebooks.py
import os, requests
API = "https://api.costkatana.com/api"
KEY = os.getenv("COST_KATANA_API_KEY")

def create_notebook(name, description=None):
    res = requests.post(f"{API}/notebooks",
        json={"name": name, "description": description},
        headers={"Authorization": f"Bearer {KEY}"})
    res.raise_for_status()
    print(f"✅ Notebook created: {res.json()['data']['id']}")
    return res.json()["data"]

def list_notebooks():
    res = requests.get(f"{API}/notebooks",
        headers={"Authorization": f"Bearer {KEY}"})
    res.raise_for_status()
    print(f"📓 Found {len(res.json()['data'])} notebooks")
    return res.json()["data"]
